packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -25,9 +25,10 @@ import type { Plugin } from '../plugin'
 
 const debug = createDebugger('vite:esbuild')
 
-// IIFE content looks like `var MyLib = function() {`. Spaces are removed when minified
+// IIFE content looks like `var MyLib = function() {`.
+// Spaces are removed and parameters are mangled when minified
 const IIFE_BEGIN_RE =
-  /(const|var)\s+\S+\s*=\s*function\(\)\s*\{.*"use strict";/s
+  /(const|var)\s+\S+\s*=\s*function\([^()]*\)\s*\{\s*"use strict";/
```

Comment:
`.*` needs to be replaced with `\s*` otherwise this test will fail.
https://github.com/vitejs/vite/blob/22bd67d70a1390daae19ca33d7de162140d533d6/playground/lib/__tests__/lib.spec.ts#L42-L49

This fix relies on the fact that https://github.com/evanw/esbuild/issues/3322 is fixed. But because that fix landed in 0.19.3, we cannot backport this fix to v4 directly.

We can use `/(const|var)\s+\S+\s*=\s*function\([^()]*\)\s*\{[\s\w,;]*"use strict";/` instead of this regex to allow `var _a, _b` part that might be injected by esbuild. But I guess it's safer to simply revert #14094 for v4.